_overviews/tutorials/binary-compatibility-for-library-authors.md

```diff
@@ -0,0 +1,221 @@
+* **Avoid breaking binary compatibility** through careful design and evolution of your library interfaces
+* Communicate binary compatibility breakages clearly through **versioning**
+
+## MiMa - Checking binary compatibility against previous library versions 
+
+The [Migration Manager for Scala](https://github.com/typesafehub/migration-manager) (MiMa) is a tool for diagnosing binary incompatibilities between different library versions.  
+It works by comparing the class files of two provided JARs and report any binary incompatibilities found. 
+Both backwards and forwards binary incompatibility can be detected by swapping input order of the JARs.
+
+By incorporating [MiMa SBT plugin](https://github.com/typesafehub/migration-manager/wiki/Sbt-plugin) into your SBT build, you can easily check whether 
+you have accidentally introduced binary incompatible changes. Detailed instruction on how to use the SBT plugin can be found in the link.
+
+We strongly encourage every library author to incorporate MiMa into their continuous integration and release workflow.
+
+Detecting backwards source compatibility is difficult with Scala due to language features like implicit
+and named parameters. The best approximation to checking backwards source compatibility is running
+both forwards and backwards binary compatibility check, as this can detect most cases
+of source-incompatible changes. For example, adding/removing public class members is a source
+incompatible change, and will be caught through forward + backward binary compatibility check.
+
+## Evolving code without breaking binary compatibility
+
+Binary compatibility breakages can often be avoided through the careful use of certain Scala features 
+as well as some techniques you can apply when modifying code.
+
+For example, the use of these language features are a common source of binary compatibility breakages
+in library releases:
+
+* Default parameter values for methods or classes
+* Case classes
+
+You can find detailed explanations, runnable examples and tips to maintain binary compatibility in [Binary Compatibility Code Examples & Explanation](https://github.com/jatcwang/binary-compatibility-guide).
+
+Again, we recommend using MiMa to double check that you have not broken binary compatibility after making changes.
+
+## Versioning Scheme - Communicating compatibility breakages
+
+Library authors use versioning schemes to communicate compatibility guarantees between library releases to their users. Versioning schemes like [Semantic Versioning](http://semver.org/)(SemVer) allow
+users to easily reason about the impact of updating a library, without needing to read the detailed release note.
+
+In the following section, we will outline a versioning scheme based on Semantic Versioning that we **strongly encourage** you to adopt for your libraries. The rules listed below are **in addition** to 
+Semantic Versioning v2.0.0.
+
+### Recommended Versioning Scheme
```

Comment:
I would also like to encourage projects to document their versioning scheme and binary compatibility rules. Can we add a short section suggesting how a project might do this? If it follows this scheme, we should recommend linking to this document in the README. If not, the README should ideally include a short section explaining the rules for that project.

Reply:
I'm trying to keep the word count down to not dilute the main takeaways of this tutorial. 

I think having another "maintaining an OSS Scala project" guide will be a good place to cover topics such as publishing to repositories and documenting versioning schemes. Speaking of which, I think it's a good idea to name this versioning scheme (e.g. "Scala Binary SemVer") instead of "that versioning scheme mentioned in that tutorial"

Reply:
I agree, while @gmethvin's suggestion is good I think it may be too much for this guide.